### PR TITLE
Add scatter functionality for loops

### PR DESCRIPTION
**Pull request by SpheMakh**
- Add pathos and dill as deps
-  set `skip_implicits = False` by default for parameters
- Add `for_loop.scatter` functionality

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,12 +9,18 @@
 requirements = ["pyyaml",
                 "nose>=1.3.7",
                 "future-fstrings",
-                "scabha @ git+https://github.com/caracal-pipeline/scabha2@master#egg=scabha-0.4.0",
+<<<<<<< HEAD
```
Comment (o-smirnov) [inline setup.py]: Looks like an unresolved merge.

File: stimela/__init__.py [hunk 2]
```diff
@@ -61,5 +61,5 @@ def register_globals():
 
 from .stimelogging import logger
 
-from stimela.kitchen.recipe import Recipe
+from stimela.kitchen.recipe import PyRecipe as Recipe
```
Comment (o-smirnov) [inline stimela/__init__.py]: Thought we're going to retire PyRecipe?

File: stimela/commands/exxec.py [hunk 3]
```diff
@@ -21,7 +21,7 @@
 
 @cli.command("exec",
     help="""
-    Execute a single cab, or a recipe from a YML file. 
+    Execute a single cab, or a recipe from a YML file_pickle.PicklingError: Can't pickle <class 'types.UpdatedStimelaConfig'>: it's not found as types.UpdatedStimelaConfig.
```
Comment (o-smirnov) [inline stimela/commands/exxec.py]: Ooops where did the error come from? Wild paste?

**Review by o-smirnov — CHANGES_REQUESTED** [review]

File: stimela/__init__.py (hunk 2 above)
Comment (sphemakh) [reply stimela/__init__.py]: fixed

File: setup.py (hunk 1 above)
Comment (sphemakh) [reply setup.py]: We can put a pypi dep before we release. But this is OK for now since using a release that doesn't exist breaks the install

**Review by o-smirnov — APPROVED** [review]

**Pull request merged**